client/src/modules/list/model/reducer.js

```diff
@@ -279,33 +293,124 @@ const lists = (state = {}, action) => {
         '_id'
       );
     }
-    case ItemActionTypes.ADD_SUCCESS:
-    case ItemActionTypes.ARCHIVE_SUCCESS:
+    case ListActionTypes.UPDATE_LIMIT: {
+      const {
+        payload: { listId, limit }
+      } = action;
+
+      const list = state[listId];
+      const previousState = list.listState;
+      const updatedState = { ...previousState, ...limit };
+
+      return {
+        ...state,
+        [listId]: {
+          ...list,
+          listState: updatedState
+        }
+      };
+    }
+    case CommentActionTypes.ADD_SUCCESS:
+    case CommentActionTypes.FETCH_SUCCESS:
     case ItemActionTypes.CLEAR_VOTE_SUCCESS:
     case ItemActionTypes.CLONE_SUCCESS:
     case ItemActionTypes.DELETE_SUCCESS:
-    case ItemActionTypes.FETCH_ARCHIVED_SUCCESS:
-    case ItemActionTypes.MARK_AS_DONE_SUCCESS:
-    case ItemActionTypes.MARK_AS_UNHANDLED_SUCCESS:
-    case ItemActionTypes.MOVE_SUCCESS:
-    case ItemActionTypes.REMOVE_ARCHIVED:
-    case ItemActionTypes.RESTORE_SUCCESS:
     case ItemActionTypes.SET_VOTE_SUCCESS:
     case ItemActionTypes.UPDATE_SUCCESS:
     case ItemStatusType.LOCK:
     case ItemStatusType.UNLOCK:
-    case CommentActionTypes.ADD_SUCCESS:
-    case CommentActionTypes.FETCH_SUCCESS: {
+    case ItemActionTypes.DISABLE_ANIMATIONS: {
+      const {
+        payload: { listId }
+      } = action;
+
+      const list = state[listId];
+
+      if (list && list.items) {
+        const { items: previousItems } = list;
+
+        return {
+          ...state,
+          [listId]: { ...list, items: items(previousItems, action) }
+        };
+      }
+
+      return state;
+    }
+    case ItemActionTypes.FETCH_ARCHIVED_SUCCESS: {
       const {
         payload: { listId }
       } = action;
 
-      if (state[listId] && state[listId].items) {
-        const { items: previousItems } = state[listId];
+      const list = state[listId];
```

Comment:
this is unnecessary, check the previous comments

Reply:
we use `list` prop in several places so as we discussed I leave `list` instead of changing to `state[listId]`